### PR TITLE
split composite example into db and app

### PR DESCRIPTION
**Pull request by csantanapr**
### What does this PR do?

Split the composite example application into AWS resources and APP

### Motivation

I was trying out the current example and was not working, app would come up and secret was not populated with values.

### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)

- [ ] Yes, I have added a new example under [examples](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/examples) to support my PR

- [ ] Yes, I have updated the [docs](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/doc) for this feature

- [ ] Yes, I have linked to an issue or feature request (applicable to PRs that solves a bug or a feature request)

**Note**:
 - Not all the PRs require examples and docs
 - We prefer small, well tested pull requests. Please ensure your pull requests are self-contained, and commits are squashed

### For Moderators
- [ ] E2E Test successfully complete before merge?

### Additional Notes

The app is hidden inside the composition with what the app was suppose to do

This PR makes the app logic visible, also the deployment is a two stage deploy the AWS resources, wait for it do be ready then deploy the app.

I could add a init container to the app, or modify the composition to add a secret observe policy to fix the race condition, but still the problem of what is the app suppose to do is hidden.

### CODE REVIEW

File: compositions/aws-provider/example-application/definition.yaml [hunk 1]
```diff
@@ -4,15 +4,15 @@
 apiVersion: apiextensions.crossplane.io/v1
 kind: CompositeResourceDefinition
 metadata:
-  name: xexampleapps.awsblueprints.io
+  name: xdynamodbexamples.awsblueprints.io
```
Comment (nabuskey) [inline compositions/aws-provider/example-application/definition.yaml:7]: I would rather not put the term DynamoDB in XRD. Reason is that this should define example applications in general, not tied to a specific type of application.

File: doc/nested-compositions.md [hunk 2]
```diff
@@ -2,53 +2,134 @@
 
 Compositions can be nested within a composition. Take a look at the example-application defined in the `compositions/aws-provider/example-application` directory. The Composition contains Compositions defined in other directories and creates a DynamoDB table, IAM policies for the table, a Kubernetes service account, and a IAM role for service accounts (IRSA). This pattern is very powerful. It let you define your abstraction based on someone else's prior work.
 
-An example yaml file to deploy this Composition is available at  `examples/aws-provider/composite-resources/example-application/example-application.yaml`.  
+An example yaml file to deploy this Composition is available at  `compositions/aws-provider/example-application/example-application-db.yaml`.  
 
 Let’s take a look at how this example application can be deployed. 
 
+## Install Composition and Composite
```
Comment (nabuskey) [inline doc/nested-compositions.md:9]: Should be `## Install Composition and Composite Resource Definitions`

**Review by nabuskey — COMMENTED** [review]
The point of this composition was to show how you can define your application (App and dependencies) as a composition in a single step. I am ok with making the documentation better but I don't think I want to separate the two.

**Comment by csantanapr** [comment]
@nabuskey 
>The point of this composition was to show how you can define your application (App and dependencies) as a composition in a single step. I am ok with making the documentation better but I don't think I want to separate the two.

The current example doesn't work, the Pod gets created before the secret `example-application-secrets` values are populated. The working example is not working. 

Couple of options if keeping the app and db together in a single composition:
1. Look into how to make the composition wait for the secrets to be populated before creating the Deployment object. I have not found a way to do this in crossplane docs.
2. Add an init container that loops/waits for the secret values to be present, then exit and let the main container run

How do you propose we move forward?

**Comment by nabuskey** [comment]
If it is a timing issue then you can do soft dependencies with patches and policies. I still think there is a value in having example applications created by compositions.
I also think a separate guide and example for deploying DB and app separately is a good idea. If you want to turn this into that, I am all for it :)

**Comment by nabuskey** [comment]
@csantanapr I am making a PR to address the issue you ran into. You should be able to see what I am doing to handle this kind of dependencies / timing issues.

**Comment by csantanapr** [comment]
>I also think a separate guide and example for deploying DB and app separately is a good idea. If you want to turn this into that, I am all for it :)

That's a good idea.
I can make this an example of deploying the app and the db claim, and then point as next step that they can use the kubernetes provider and have a claim that encapsulate both the app and the db with a single claim.

**Comment by nabuskey** [comment]
Closing this for now. Feel free to reopen.

**Pull request closed without merging**